### ISSUE TITLE
[BUG] can't type Chinese on Safari

### ISSUE THREAD

**Issue by BennyThink** [Ajaxy/telegram-tt#272]
Chrome and Firefox works fine, bu not for Safari. See attach video for detail

Safari Version 16.5.2 (18615.2.9.11.10)
macOS Ventura 13.4.1 (c) 

Telegram Web A 10.0.0

EDIT: it works fine before today's update

https://github.com/Ajaxy/telegram-tt/assets/14024832/9e58c106-056c-47b5-b117-88d094da0259

**Comment by BennyThink** [comment]
it seems this file `src/util/focusEditableElement.ts` 
```ts
  if (!IS_TOUCH_ENV && !forcePlaceCaretAtEnd && (!lastChild || !lastChild.nodeValue)) {
    requestMeasure(() => element.focus());
    return;
  }
```

`requestMeasure` is incompatible with Safari. If I change it back to `element.focus()` then everything works fine.
Or `!IS_TOUCH_ENV && !IS_SAFARI &&...` to make an exemption for Safari.
 
Do you want a PR for it? Or fix it directly since it's two lines of code.

**Comment by Ajaxy** [comment]
Hello, can you check if changing `requestMeasure → forceMeasure` works?

**Comment by wangw469** [comment]
> Hello, can you check if changing `requestMeasure → forceMeasure` works?

I can reproduce the problem, and I want to try if requestMeasure → forceMeasure works, but I can't get a valid app id from https://my.telegram.org/apps.

After submit the form, always got error, can you please tell me how to get a app id ? Thanks

<img width="405" alt="Screenshot 2023-08-24 at 7 59 23 AM" src="https://github.com/Ajaxy/telegram-tt/assets/3417025/38f68ac3-c2e8-4cb1-a779-b4f39700c7c7">

<img width="441" alt="Screenshot 2023-08-24 at 8 00 34 AM" src="https://github.com/Ajaxy/telegram-tt/assets/3417025/777ddee3-c4be-4ff9-b8c0-fbc113c8e11a">

**Comment by BennyThink** [comment]
Sure, I'll check it tonight and get back to you

**Comment by BennyThink** [comment]
Sadly no.
```ts
import { requestMeasure, forceMeasure } from "../lib/fasterdom/fasterdom";
.....
  if (!IS_TOUCH_ENV && !IS_SAFARI && !forcePlaceCaretAtEnd && (!lastChild || !lastChild.nodeValue)) {
    forceMeasure(() => element.focus());
    return;
  }

```


Error traceback
```
ERROR
The current phase is 'measure'
forceMeasure@http://localhost:1234/main.9eff3f8863b471f912aa.js:19731:20
focusEditableElement@http://localhost:1234/src_components_common_Audio_tsx-src_components_common_AvatarList_tsx-src_components_common_Ch-2ffb91.e02e99f3be186cb67f16.js:19119:75
handleChange@http://localhost:1234/src_components_common_Audio_tsx-src_components_common_AvatarList_tsx-src_components_common_Ch-2ffb91.e02e99f3be186cb67f16.js:10556:79
handleEvent@http://localhost:1234/main.9eff3f8863b471f912aa.js:20217:16
```

<img width="487" alt="image" src="https://github.com/Ajaxy/telegram-tt/assets/14024832/e6d700a4-2a7d-4f89-9127-ebcb68de4627">
<img width="715" alt="image" src="https://github.com/Ajaxy/telegram-tt/assets/14024832/aa4f522e-a733-46e4-93a6-91384a6db19b">

**Comment by wangw469** [comment]
> Sadly no.
> 
> ```ts
> import { requestMeasure, forceMeasure } from "../lib/fasterdom/fasterdom";
> .....
>   if (!IS_TOUCH_ENV && !IS_SAFARI && !forcePlaceCaretAtEnd && (!lastChild || !lastChild.nodeValue)) {
>     forceMeasure(() => element.focus());
>     return;
>   }
> ```
> 
> Error traceback
> 
> ```
> ERROR
> The current phase is 'measure'
> forceMeasure@http://localhost:1234/main.9eff3f8863b471f912aa.js:19731:20
> focusEditableElement@http://localhost:1234/src_components_common_Audio_tsx-src_components_common_AvatarList_tsx-src_components_common_Ch-2ffb91.e02e99f3be186cb67f16.js:19119:75
> handleChange@http://localhost:1234/src_components_common_Audio_tsx-src_components_common_AvatarList_tsx-src_components_common_Ch-2ffb91.e02e99f3be186cb67f16.js:10556:79
> handleEvent@http://localhost:1234/main.9eff3f8863b471f912aa.js:20217:16
> ```
> 
> <img alt="image" width="487" src="https://user-images.githubusercontent.com/14024832/263070323-e6d700a4-2a7d-4f89-9127-ebcb68de4627.png"> <img alt="image" width="715" src="https://user-images.githubusercontent.com/14024832/263070467-aa4f522e-a733-46e4-93a6-91384a6db19b.png">

Yes, I can confirm that forceMeasure doesn't work for safari either, with same error message.

**Comment by BennyThink** [comment]
Closing this as this was fixed under new version of Safari Version 17.6 (19618.3.11.11.5) and macOS 14.6.1 (23G93)
